src/data/metadata.ts

```diff
@@ -54,32 +62,39 @@ function guid() {
   let uuid = "xxxxxxxxxxxx4xxxyxxxxxxxxxxxxxxx".replace(/[xy]/g, function(c) {
     let r = (d + Math.random() * 16) % 16 | 0;
     d = Math.floor(d / 16);
-    return (c == "x" ? r : (r & 0x3 | 0x8)).toString(16);
+    return str((c == "x" ? r : (r & 0x3 | 0x8)), 16);
   });
   return uuid;
 }
 // tslint:enable
 
-function track(data: CookieInfo): void {
-  let expiry = new Date();
-  expiry.setDate(expiry.getDate() + config.expire);
-  let expires = expiry ? "expires=" + expiry.toUTCString() : "";
-  let value = `${data.userId}|${data.sessionId}|${data.timestamp}` + ";" + expires + ";path=/";
-  document.cookie = CLARITY_COOKIE_NAME + "=" + value;
+function session(ts: number): string {
+  let id = str(ts, 36);
+  if (config.track && sessionStorage) {
+    let value = sessionStorage.getItem(CLARITY_STORAGE_KEY);
+    if (value && value.indexOf(CLARITY_STORAGE_SEPARATOR) >= 0) {
+      let parts = value.split(CLARITY_STORAGE_SEPARATOR);
+      if (parts.length === 2 && ts - parseInt(parts[1], 10) < config.session) { id = parts[0]; }
+    }
+    sessionStorage.setItem(CLARITY_STORAGE_KEY, `${id}|${ts}`);
```

Comment:
>} [](start = 53, length = 1)

Should use the STORAGE_SEPARATOR here. It's hardcoded in the string.

Reply:
Thanks, will fix.